docs/docs/code/go/README.md

```diff
@@ -1,6 +1,452 @@
+  log.Println(resp.Status)
+
+  // The response is logged in your Pipedream step results:
+  sb := string(body)
+  log.Println(sb) 
+}
+```
+
+### Making a POST request
+
+```go
+package main
+
+import (
+   "bytes"
+   "encoding/json"
+   "io/ioutil"
+   "log"
+   "net/http"
+)
+
+func main() {
+  // JSON encode our payload
+   payload, _ := json.Marshal(map[string]string{
+      "name":  "Bulbasaur",
+   })
+   payloadBuf:= bytes.NewBuffer(payload)
+
+  // Send the POST request
+   resp, err := http.Post("https://postman-echo.com/post", "application/json", payloadBuf)
+
+   if err != nil {
+      log.Fatalln(err)
+   }
+   defer resp.Body.Close()
+
+  //Read the response body
+   body, err := ioutil.ReadAll(resp.Body)
+   if err != nil {
+      log.Fatalln(err)
+   }
+   // Convert the body into a string
+   sb := string(body)
+   // Log the body to our Workflow Results
+   log.Println(sb)
+}
+```
+
+### Sending files
+
+You can also send files within a step.
+
+An example of sending a previously stored file in the workflow's `/tmp` directory: 
+
+```go
+package main
+
+import (
+  "os"
+  "log"
+  "mime/multipart"
+  "bytes"
+  "io"
+  "mime/multipart"
+  "net/http"
+  "os"
+)
+
+func main() {
+  // Instantiate a new HTTP client, body and form writer
+  client := http.Client{}
+  body := &bytes.Buffer{}
+  writer := multipart.NewWriter(body)
+
+  // Create an empty file to start
+  fw, err = writer.CreateFormFile("file", "go-logo.svg")
+  if err != nil {
+    log.Fatalln(e)
+  }
+
+  // Retrieve a previously saved file from workflow storage
+  file, err := os.Open("/tmp/go-logo.svg")
+  if err != nil {
+    log.Fatalln(e)
+  }
+
+  // Close multipart form writer
+  writer.Close()
+
+  _, err = io.Copy(fw, file)
+
+  // Send the POST request
+  req, err := http.NewRequest("POST", "https://postman-echo.com/post", bytes.NewReader(body.Bytes()))
+  if e != nil {
+    log.Fatalln(e)
+  }
+
+  req.Header.Set("Content-Type", writer.FormDataContentType())
+  rsp, err := client.Do(req)
+  if e != nil {
+    log.Fatalln(e)
+  }
+}
+```
+
+## Sharing data between steps
+
+A step can accept data from other steps in the same workflow, or pass data downstream to others.
+
+This makes your steps even more powerful, you can compose new workflows and reuse steps.
+
+### Using data from another step
+
+In Go steps, data from the initial workflow trigger and other steps are available in the `pipedream-go` package.
+
+In this example, we'll pretend this data is coming into our HTTP trigger via POST request.
+
+```json
+{
+  "id": 1,
+  "name": "Bulbasaur",
+  "type": "plant"
+}
+```
+
+In our Go step, we can access this data in the `Steps` variable from the `pd` package. Specifically, this data from the POST request into our workflow is available in the `trigger` dictionary item. 
+
+```go
+package main
+
+import (
+	"fmt"
+	"github.com/PipedreamHQ/pipedream-go"
+)
+
+func main() {
+	// Access previous step data using pd.Steps
+	fmt.Println(pd.Steps)
```

Comment:
Since you referenced the `trigger` step, maybe show the user how to access that specifically:

```go
fmt.Println(pd.Steps["trigger"])
```